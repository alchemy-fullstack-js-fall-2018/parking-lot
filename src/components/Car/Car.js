import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import ROUTES from '../../routes';


const Car = ({ car }) => {
  const { plate, make, model, color } = car;
  return (
    <div>
      <p><Link to={ROUTES.CAR.linkTo(plate)}>{plate}</Link>: {color} {make} {model} </p>
    </div>
  );
};

Car.propTypes = {
  car: PropTypes.object.isRequired
};

export default Car;
