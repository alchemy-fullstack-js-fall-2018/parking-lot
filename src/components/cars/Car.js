import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';

const Car = ({ cars }) => {

  const carItems = cars.map(car => {
    return (
      <tr key={car.plate}>
        <td id="make">{car.make}</td>
        <td id="model">{car.model}</td>
        <td id="plate"><Link to={ROUTES.CAR_DETAIL.linkTo(car.plate)}>{car.plate}</Link></td>
      </tr>
    );
  });
  

  return (
    <Fragment>
      {carItems}
    </Fragment>
  );
};

Car.propTypes = {
  cars: PropTypes.array.isRequired
};

export default Car;
