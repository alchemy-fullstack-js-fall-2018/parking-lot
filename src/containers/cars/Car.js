import { connect } from 'react-redux';
import Car from '../../components/Car/Car';
import { getCar } from '../../selectors/cars';

const mapStateToProps = (state, props) => {
  return {
    car: getCar(state, props.match.params.id)
  };
};

export default connect(
  mapStateToProps
)(Car);
