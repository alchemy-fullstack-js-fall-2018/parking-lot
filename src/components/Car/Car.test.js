import React from 'react';
import { shallow } from 'enzyme';
import Car from './Car';

describe ('tests for Car component', () => {

  it('renders a Car component', () => {
    const car = {
      plate: 'ZZZ000',
      make: 'Plymouth',
      model:  'Edsel',
      color: 'pink'
    };
    const wrapper = shallow(<Car car={car}/>);
    expect(wrapper).toMatchSnapshot();
  });

});
