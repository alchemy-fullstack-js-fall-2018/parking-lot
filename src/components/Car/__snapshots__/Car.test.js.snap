// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for Car component renders a Car component 1`] = `
<div>
  <p>
    <Link
      replace={false}
      to="/cars/ZZZ000"
    >
      ZZZ000
    </Link>
    : 
    pink
     
    Plymouth
     
    Edsel
     
  </p>
</div>
`;
